### PR TITLE
fix(frontend): do not show token logo in token transaction page

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation (deprecated, see Edit)

I don't have a better idea: the token logo was being showed in token-specific transaction page, because it is given as input and it is required in components like `BtcTransaction` or `EthTransaction`.

However, that same component is needed for the unified transactions (example PR https://github.com/dfinity/oisy-wallet/pull/3570) with the token logo.

So, the only idea I had was to make it so by default the component show the token logo, but not under request (for example in the token-specific transaction page).

# Edit

After @peterpeterparker [suggestion](https://github.com/dfinity/oisy-wallet/pull/3579#discussion_r1844114080) , I changed the logic to provide a prop that defines the icon type (either `token` or `transaction`). In my opinion is a better solution.

### CODE REVIEW

File: src/frontend/src/lib/components/transactions/Transaction.svelte [hunk 1]
```diff
@@ -31,7 +32,7 @@
 		<span class="inline-block first-letter:capitalize"><slot /></span>
 
 		<div slot="icon">
-			{#if nonNullish(token)}
+			{#if nonNullish(token) && !hideTokenLogo}
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/transactions/Transaction.svelte:35]: The token is mandatory now, I guess we can remove `nonNullish(token)`?

File: src/frontend/src/lib/components/transactions/Transaction.svelte [hunk 2]
```diff
@@ -18,6 +18,7 @@
 	export let timestamp: number | undefined;
 	export let styleClass: string | undefined = undefined;
 	export let token: Token;
+	export let hideTokenLogo = false;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/transactions/Transaction.svelte:21]: Another option is something like as following

```
export let badge: "token" | "logo" = "logo";
```

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: src/frontend/src/lib/components/transactions/Transaction.svelte (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/transactions/Transaction.svelte:35]: make sense, done!

File: src/frontend/src/lib/components/transactions/Transaction.svelte (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/transactions/Transaction.svelte]: way better!!! thank you! changed to iconType! have a look!

**Comment by AntonioVentilii** [comment]
@peterpeterparker even if you already approved, please re-review, since I followed [your suggestion](https://github.com/dfinity/oisy-wallet/pull/3579#discussion_r1844114080) and changed the logic

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Comment by AntonioVentilii** [comment]
[OISY-130]

**Pull request merged**